yb-voyager/cmd/common.go

```diff
@@ -637,35 +719,141 @@ func initBaseTargetEvent(bev *cp.BaseEvent, eventType string) {
 	}
 }
 
-func renameTableIfRequired(table string) string {
+func renameTableIfRequired(table string) (string, bool) {
 	// required to rename the table name from leaf to root partition in case of pg_dump
 	// to be load data in target using via root table
 	msr, err := metaDB.GetMigrationStatusRecord()
 	if err != nil {
 		utils.ErrExit("Failed to get migration status record: %s", err)
 	}
-	source = *msr.SourceDBConf
-	if source.DBType != POSTGRESQL {
-		return table
+	sourceDBType = msr.SourceDBConf.DBType
+	sourceDBTypeInMigration := msr.SourceDBConf.DBType
+	schema := msr.SourceDBConf.Schema
+	sqlname.SourceDBType = source.DBType
+	if source.DBType != POSTGRESQL && source.DBType != YUGABYTEDB {
+		return table, false
+	}
+	if source.DBType == POSTGRESQL && msr.SourceRenameTablesMap == nil ||
+		source.DBType == YUGABYTEDB && msr.TargetRenameTablesMap == nil {
+		return table, false
 	}
-	if msr.RenameTablesMap == nil {
-		return table
+	if sourceDBTypeInMigration != POSTGRESQL && source.DBType == YUGABYTEDB {
+		schema = source.Schema
 	}
-	defaultSchema, noDefaultSchema := GetDefaultPGSchema(source.Schema, "|")
+	renameTablesMap := msr.SourceRenameTablesMap
+	if source.DBType == YUGABYTEDB {
+		renameTablesMap = msr.TargetRenameTablesMap
+	}
+	defaultSchema, noDefaultSchema := GetDefaultPGSchema(schema, "|")
 	if noDefaultSchema && len(strings.Split(table, ".")) <= 1 {
 		utils.ErrExit("no default schema found to qualify table %s", table)
 	}
 	tableName := sqlname.NewSourceNameFromMaybeQualifiedName(table, defaultSchema)
 	fromTable := tableName.Qualified.Unquoted
```

Comment:
Yeah. You were using `.Qualified.Unquoted` so I was wondering...

Reply:
Hmm, Actually this is because the map I create for leaf->root mapping is also used by debezium where debezium only deals with `Unquoted` names even for case sensitivity so had to create it like that.